File: 백준/Bronze/1292. 쉽게 푸는 문제/쉽게 푸는 문제.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {

	public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
		ArrayList<Integer> arr = new ArrayList<Integer>();

		int num = 1;
		int count = 0;
    
		for (int i = 0; i < 1000; i++) {
			arr.add(num);
			count++;
			if (num == count) {
				num++;
				count = 0;
			}
		}

		int num1 = s.nextInt();
		int num2 = s.nextInt();
		int sum = 0;
		for (int i = num1 - 1; i <= num2 - 1; i++) {
			sum += arr.get(i);
		}
		System.out.println(sum);
	}

}

File: 백준/Bronze/1292. 쉽게 푸는 문제/쉽게 푸는 문제.cc
#include <iostream>
#include <vector>
using namespace std;


int main () {
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);

    vector<int> arr;
    int num = 1;

    for (int i = 0; i < 1000; i++) {
        for (int j = 1; j <= num; j++) {
            arr.push_back(num);
        }
        num++;
    }

    int start, end, sum = 0;
    cin >> start >> end;

    for (int i = start-1; i <= end-1; i++) {
        sum += arr[i];
    }

    cout << sum;

    return 0;
}  